### ISSUE TITLE
Unable to get working with HTTP Proxy

### ISSUE THREAD

**Issue by matttrawicki** [crops/chameleonsocks#3]
Hello! I'm trying to get this working as part of the Yocto/Ostro build system on Ubuntu Server 16.04.1 with my company's HTTP proxy.

chameleonsocks.sh settings are as follows:

``` bash
: ${PROXY:=http://gateway.zscalertwo.net}
: ${PORT:=12345}
: ${PROXY_TYPE:=http-connect}
: ${EXCEPTIONS:=/usr/local/etc/chameleonsocks.exceptions}
```

My exceptions file is blank (for now). I get a notification during install about how it's going to load the defaults.

Simple curl command fails:

```
$ curl http://www.microsoft.com -v -4
* Rebuilt URL to: http://www.microsoft.com/
*   Trying 23.73.78.60...
* connect to 23.73.78.60 port 80 failed: Connection refused
* Failed to connect to www.microsoft.com port 80: Connection refused
* Closing connection 0
curl: (7) Failed to connect to www.microsoft.com port 80: Connection refused
```

Docker log says:

```
me@computer:~$ sudo docker logs chameleonsocks
sed: -e expression #1, char 24: unknown option to `s'
iptables: Chain already exists.
curl: try 'curl --help' or 'curl --manual' for more information
file parsing error at line 30: Name or service not known
file parsing error at line 30: value can't be parsed
file parsing error at line 30: unclosed section
```

I'm used to setting up the config file for every tool as well as the env vars but liked the idea of just setting it up in one place. Thanks for your help!

**Comment by todorez** [comment]
Hi,
Congratulations, you have just found a bug! :). I think I see what the problem is. 
- Let's uninstall chameleonsocks so we can start with a fresh environment.

`./chameleonsocks.sh --uninstall
`
- Try to use these settings in the install script (chameleonsocks.sh).

```
: ${PROXY:=gateway.zscalertwo.net}
: ${PORT:=12345}
: ${PROXY_TYPE:=http-connect}
: ${EXCEPTIONS:=/usr/local/etc/chameleonsocks.exceptions}
: ${PAC_URL:=http://www.example.com}
```

Hopefully this will work better for you.

**Comment by matttrawicki** [comment]
Thank you for jumping on my problem. What's the bug? Please help me understand your suggestions.
1. Strip the protocol off the PROXY url. That's easy, but to be clear, my proxy does not have an HTTPS endpoint. Does that matter? The README example does not include the protocol portion of the url, so I did try that before writing up this issue. But I'll make sure to omit it going forward.
2. PAC_URL. Hmm. I left this commented out because I do not have a PAC file, this is a direct proxy.

**Comment by todorez** [comment]
1. Redsocks requires only the domain name(or ip address) of your proxy server without specifying a protocol type. We can be smarter about this and strip any protocol prefix from the PROXY url if it is provided. I will open issue to track this.
2. The bug is that we have assumed that PAC_URL will be set to something when in practice the user does not have to provide a PAC server at all. For the moment you can provide a dummy value for PAC_URL( e.g. www.example.com) and I will get this fixed in the next release.

**Comment by matttrawicki** [comment]
Thank you. That got rid of all of the errors in my docker logs. I'm having some other unrelated networking issues so I'm not 100% up and running yet. I'll leave this issue open to track the pac file required bug?

**Comment by todorez** [comment]
Closing. Fixed with 1f9da7670cb401e82fd07b4fc4941fbedc1d85a7
